File: 080__Remove_Duplicates_from_Sorted_Array_II.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        length = 1
        cur_val, cur_num = nums[0], 1
        for i in range(1, len(nums)):
            if nums[i] == cur_val:
                cur_num += 1
                if cur_num == 2:
                    nums[length] = nums[i]
                    length += 1
            else:
                cur_val = nums[i]
                cur_num = 1
                nums[length] = nums[i]
                length += 1
        return length

File: Java/080__Remove_Duplicates_from_Sorted_Array_II.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int idx = 1, count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1]) {
                if (count == 2) {
                    continue;
                } else {
                    count++;
                    nums[idx] = nums[i];
                    idx++;
                }
            } else {
                count = 1;
                nums[idx] = nums[i];
                idx++;
            }
        }
        return idx;
    }
}
